### PR TITLE
[MS] Delete & set default payment method

### PR DESCRIPTION
**Pull request by Max-7**
- [X] Keep changes in the pull request as small as possible
- [X] Ensure the commit history is sanitized
- [X] Give a meaningful title to your PR
- [X] Describe your changes

### CODE REVIEW

File: client/src/locales/fr-FR.json [hunk 1]
```diff
@@ -1347,22 +1347,51 @@
...
-            "addPaymentMethodTitle": "Ajouter une carte bancaire",
-            "addPaymentMethodConfirm": "Ajouter cette carte",
+            "description": "Le paiement s'effectue automatiquement par votre carte bancaire.",
             "useAsDefault": "Définir comme moyen de paiement actif",
-            "activeMethodLabel": "Active",
+            "activeMethodLabel": "active",
+            "activeDefault": "Carte utilisée par défaut",
             "retrieveFailed": "Impossible de récupérer les informations de paiement.",
-            "addPaymentMethodSuccess": "La carte a été ajoutée avec succès.",
+            "savedCard": "Cartes enregistrées",
```
Comment (NicoTuxx) [inline client/src/locales/fr-FR.json:1355]: missing translations `setDefault` and `delete`
image

**Review by NicoTuxx — CHANGES_REQUESTED** [review]
- [x] When adding a new card, `set as default payment method` does refresh the card background, but not the card details.
https://github.com/user-attachments/assets/d1cbe110-fdf7-4399-8f14-0c7304833c56
- [ ] When deleting a card, an undesired skeleton-text is appearing.
https://github.com/user-attachments/assets/6919e8a5-08be-4f06-a2b9-84ccd5986ede
- [ ] When using `Set as default` it is combining the 2 previous bugs.
https://github.com/user-attachments/assets/72141646-7714-4990-93aa-00fd40a3111d
- [ ] When having an error on `Add new card`, missing translation (fr & en) (with and without 'set default').
- [ ] When having an error on `Add new card`, should not close the modal, force to restart whole process.
https://github.com/user-attachments/assets/f98d5257-3c30-4f75-a29f-a9eda9e7bdeb
- [ ] Missing checks on `Set default` button and API: a card can expired, we should hide the button in this case and at quickly as possible to leave the function if card is expired.
- [ ] Missing check on `Delete payment method` API function, it should check if the card is not the active one (even if the UX does not offer such possibility).
- [ ] Is the `Restart subscription` supposed to be working? I don't remember it exisiting in the actual merchant site, and here I've an error each time I tried it.
image

File: client/src/locales/en-US.json [hunk 2]
```diff
@@ -1348,20 +1348,52 @@
...
+            },
             "stopSubscription": {
                 "title": "Would you like to stop your subscription?",
-                "description": "The organisation will become unavailable to all users.",
-                "button": "Stop my subscription"
-            }
+                "description": "The organization will become unavailable to all users.",
+                "button": "Stop my subscription",
+                "questionTitle": "Stop the subscription",
+                "questionMessage": "Are you sure you want to the subscription?",
```
[the remark this answers is not in the document]
Comment (NicoTuxx) [reply client/src/locales/en-US.json:1370]: done

File: client/src/locales/fr-FR.json [hunk 3]
```diff
@@ -1347,22 +1347,51 @@
...
+            },
+            "restartSubscription": {
+                "title": "Vous souhaitez reprendre votre abonnement ?",
+                "description": "Votre organisation sera à nouveau disponible.",
+                "button": "Reprendre mon abonnement",
+                "questionTitle": "Reprendre l'abonnement",
+                "questionMessage": "Souhaitez-vous reprendre votre abonnement ?",
+                "failed": "Impossible de reprendre l'abonnement.",
+                "success": "L'abonnement a redémarré.",
+                "confirm": "reprendre mon abonnement",
```
[the remark this answers is not in the document]
Comment (NicoTuxx) [reply client/src/locales/fr-FR.json:1382]: done

File: client/src/locales/fr-FR.json (hunk 1 above)
Comment (NicoTuxx) [reply client/src/locales/fr-FR.json:1355]: done

**Comment by Max-7** [comment]
* [X]  When having an error on `Add new card`, should not close the modal, force to restart whole process.
* [X]  Missing checks on `Set default` button and API: a card can expired, we should hide the button in this case and at quickly as possible to leave the function if card is expired.

* [ ]  Missing check on `Delete payment method` API function, it should check if the card is not the active one (even if the UX does not offer such possibility).

Does it matter? There's no button for it in the interface, and the BMS is the arbiter in this case, it will decide if it accepts it or not (and return an error).

* [X]  Is the `Restart subscription` supposed to be working? I don't remember it exisiting in the actual merchant site, and here I've an error each time I tried it.

It does, but it's per organization and I forgot to hide it we were in the `All organization` case, meaning it was trying to hit the route `**/organizations//subscribe` because it was missing the org id.  That's probably the case you encountered.

**Review by NicoTuxx — APPROVED** [review]

**Pull request merged**